build/src/lib.rs

```diff
@@ -369,11 +352,17 @@ impl BinaryBuilder {
                 assert_ne!(transfer_buffer_rules.len(), 0);
                 assert_ne!(transfer_buffer_tokenizer.len(), 0);
 
-                transform_data_fn(Box::new(&mut transfer_buffer_rules.as_slice()), Box::new(rules_sink))
-                    .map_err(Error::TransformError)?;
+                let mut transformed_buffer_rules = Vec::new();
+                let mut transformed_buffer_tokenizer = Vec::new();
+
+                transform_data_fn(
+                    Cursor::new(transfer_buffer_rules),
+                    Cursor::new(&mut transformed_buffer_rules),
+                )
+                .map_err(Error::TransformError)?;
                 transform_data_fn(
```

Comment:
Not sure if the cursors are really needed anymore, since the `Vec<u8>` references are also `impl Read` and `impl Write`, so the closures could work just fine in that case.

Reply:
Good point! They really weren't needed anymore since `&[u8]` implements `Read` and `&mut Vec<u8>` implements `Write`. Now the signature is much nicer and `Cursor`s aren't needed anymore :)